libethereum/EthereumPeer.cpp

```diff
@@ -304,6 +307,11 @@ bool EthereumPeer::interpret(unsigned _id, RLP const& _r)
 			break;
 		}
 		setIdle();
+		if (itemCount > c_maxIncomingHashes)
+		{
+			disable("Too many hashes");
```

Comment:
future PR might check simply whether the `itemCount` is at most the number we actually asked for.